File: algorithm-py/cn/sort/Selection.py
'''
Created on 2016-6-11

@author: sonn
'''
from cn.sort.Base import Base

class Selection(Base):
    
    def sort(self,arr):
        arrLen = len(arr)
        for i in range(0,arrLen-1):
            min = i
            for j in range(i + 1, arrLen - 1):
                if(self.less(arr[j], arr[min]) < 0):
                    min = j
            self.exch(arr, i, min)

            

File: algorithm-java-demo/src/cn/sort/Selection.java
package cn.sort;

/** 
* @ClassName: Selection 
* @Description: 选择排序类 
* @author 无名
* @date 2016-6-9 下午7:23:59 
* @version 1.0
 */

public class Selection extends Base
{
	@Override
	public void sort(Comparable[] arr)
	{	
		int arrLength = arr.length;
		for (int i = 0; i < arrLength; i++)
		{
			int min = i;
			for(int j = i + 1; j < arrLength; j++)
			{
				if(less(arr[j], arr[min])) 
				{
					min = j;
				}
			}
			exch(arr, i, min);
		}
	}
}
